### ISSUE TITLE
dx new command fails - v0.6.0  

### ISSUE THREAD

**Issue by elewis787** [DioxusLabs/dioxus#3319]
**Problem**

When calling `dx new example` the cli fails with 

```
dx new hayride-ui
   1.317s ERROR err=Other(Please check if the Git user / repository exists.

Caused by:
    could not read from remote repository; class=Net (12); code=Eof (-20))
```

I have validated by git config user name and auth. 

**Steps To Reproduce**

- install v0.6.0
  - followed the docs to install and blog for v0.6.0 using cargo binstall 

**Expected behavior**
Create a new dioxus repo from template repo 

**Environment:**
 - Dioxus version:  cli v0.6.0 
 - Rust version:    cargo 1.84.0-nightly (031049782 2024-11-01)
 - OS info:     MacOS

**Comment by elewis787** [comment]
falling back to v0.5.0 of the cli greats around the above error.

**Comment by cloudbridgeuy** [comment]
I'm having the same issue.

Rust version: `rustc 1.83.0 (90b35a623 2024-11-26)`
Os info: macOS (ARM)
Dioxus version: `dioxus 0.6.0 (3e1e7f0)`

**Comment by emile2013** [comment]
so am i, some versions:
Rust version: rustc 1.83.0 (90b35a623 2024-11-26)
Os info: macOS (ARM)
Dioxus version: dioxus  0.6.1

**Comment by hoel-bagard** [comment]
I had the same issue, but by removing the following from my git config it went away.

```
[url "git@github.com:"]
	insteadOf = "https://github.com/"
````

**Comment by ealmloff** [comment]
> I'm curious now, why does it get from git. Templates?

Yes, the templates are pulled from https://github.com/DioxusLabs/dioxus-template/tree/v0.6

**Comment by wojciechrak** [comment]
With 0.5.7 I still cannot use templates
```sh
 dx new example --template=web
Error: 🚫 Creating new project failed:

Caused by:
    0: Please check if the Git user / repository exists.
    1: unsupported URL protocol; class=Net (12)
```

My .gitconfig consists only of user name and email

**Comment by mayathomas** [comment]
You can check your network proxy.

**Comment by dastoc** [comment]
The error persists on ❯ dioxus 0.6.3 (fc1f1c2).
Any update?

**Comment by aaronlifton** [comment]
Error is still occurring on multiple machines, without a proxy 🤔

**Comment by lgy87** [comment]
same issue here.
by removing ~/.gitconfig it went away.

`mv ~/.gitconfig{,.bak}`

**Comment by julianbolts** [comment]
thanks, @lgy87 's fix worked for me. unfortunately, this is a terrible DX. was this fixed in the upcoming 0.7.x tree?
p.s. - if you used Igy's command, don't forget to revert the file rename:)

**Comment by darklajid** [comment]
Same problem, different usecase: I have a combination of ssh configuration and git configuration set up to use different ssh keys and my .gitconfig contains, among other things,

```
[url "ssh://git@gh_dar:"]
    insteadOf = https://github.com
```

which ssh knows how to handle. It breaks literally the first step in the DX experience though with a nasty panic (0.6.3 here)

**Comment by cylixlee** [comment]
Same issue here. The `url.insteadOf` is very useful and I don't want to change my `git clone` habit -- I just copy the URL and paste it to `git clone` 😂.

I wonder whether this issue could be solved without removing my Git configuration?
